commit 5309a019756c479a3508373dd157f5964761b8e9 (refs/changes/69/269/3)
Author: Marisa Emerson <mje@uq.edu.au>
Date:   2016-08-11T10:26:03-07:00 (3 years, 2 months ago)
    
    joyent/triton-cns#3 network wildcard entries doesn't work on multiple zones
    Reviewed by: Alex Wilson <alex.wilson@joyent.com>

diff --git a/lib/net-filter.js b/lib/net-filter.js
index 52a3a21..df51975 100644
--- a/lib/net-filter.js
+++ b/lib/net-filter.js
@@ -69,11 +69,12 @@ NetFilter.prototype._transform = function (vm, enc, cb) {
 				}
 			}
 		}
-		for (i = 0; i < zones.length; ++i) {
-			z = self.config[zones[i]];
-			if (z.networks.indexOf('*') !== -1 &&
-			    nic.zones.length === 0)
-				nic.zones.push(zones[i]);
+		if (nic.zones.length === 0) {
+			for (i = 0; i < zones.length; ++i) {
+				z = self.config[zones[i]];
+				if (z.networks.indexOf('*') !== -1)
+					nic.zones.push(zones[i]);
+			}
 		}
 	});
 
diff --git a/test/unit/net-filter.test.js b/test/unit/net-filter.test.js
index 4235ce7..69da062 100644
--- a/test/unit/net-filter.test.js
+++ b/test/unit/net-filter.test.js
@@ -133,6 +133,29 @@ test('wildcard does not match if an exact is present after', function (t) {
 	t.end();
 });
 
+test('multiple wildcards', function (t) {
+	var config = {
+		bar: {
+			networks: ['*']
+		},
+		foo: {
+			networks: ['abc123']
+		},
+		derp: {
+			networks: ['*']
+		}
+	};
+	var s = new NetFilter({config: {forward_zones: config}});
+	s.write({
+		uuid: 'abcd1234',
+		nics: [ { network_uuid: 'def123' } ]
+	});
+	var out = s.read();
+	t.strictEqual(typeof (out), 'object');
+	t.deepEqual(out.nics[0].zones, ['bar', 'derp']);
+	t.end();
+});
+
 test('refuses admin network', function (t) {
 	var config = {
 		foo: {
